src/functions-and-arrays.js

```diff
@@ -1,19 +1,148 @@
 // Iteration #1: Find the maximum
 
+const maxOfTwoNumbers = (number1, number2) => {
+  if (typeof number1 !== 'number' || typeof number2 !== 'number') {
+    return 'Please introduce 2 numbers';
+  }
+  
+  if (number1 > number2) {
+    return number1
+  } else {
+    return number2;
+  }
+};
+
+maxOfTwoNumbers();
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+const findLongestWord = arrayOfWords => {
+  let characters = 0;
+  let longestWord = '';
+
+  for(let i = 0; i < arrayOfWords.length; i++) {
+    if (arrayOfWords[i].length > characters){
+      characters = arrayOfWords[i].length;
+      longestWord = i;
+    }
+  }
+
+  return arrayOfWords[longestWord];
+}
+
+findLongestWord(words);
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+const sumNumbers = arrayOfNumbers => {
+  let total = 0;
+
+  for(let i = 0; i < arrayOfNumbers.length; i++) {
+    total += arrayOfNumbers[i];
+  }
+
+  return total;
+}
+
+sumNumbers(numbers);
+
+//#### Bonus - Iteration #3.1: A generic `sum()` function
+
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+
+const sum = anyArray => {
+  let total = 0;
+
+  for(let i = 0; i < anyArray.length; i++) {
+
+    switch (typeof anyArray[i]) {
+      case 'string':
+        total += anyArray[i].length;
+        break;
+      case 'number':
+        total += anyArray[i];
+        break;
+      case 'boolean':
+        total += anyArray[i];
+        break;
+      default:
+        return 'Objects or Arrays inside the Array are not permited, only numbers, strings and boolean.';
+    }
+  }
+
+  return total;
+}
+
+sum(mixedArr);
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
+
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+const averageNumbers = arrayOfNumbers => {
+  let total = 0;
+
+  for(let i = 0; i < arrayOfNumbers.length; i++) {
+    total += arrayOfNumbers[i];
+  }
+
+  return Math.round((total / arrayOfNumbers.length) * 10) / 10;
+}
+
```

Comment:
You could shorten your code if you made use of your `sumNumbers` function

`return Math.round(sumNumbers(arrayOfNumbers)/arrayOfNumbers.length * 10)/10`